.github/workflows/ci.yml

```diff
@@ -38,117 +37,82 @@ jobs:
+        include:
+          # Linux x86
+          - runs-on: ubuntu-latest
+            backend: vulkan
+          - runs-on: ubuntu-latest
+            backend: opengl
+          # Linux ARM
+          - runs-on: ubuntu-24.04-arm
+            backend: opengl
+          # macOS ARM
+          #- runs-on: macos-latest
+          #  backend: metal
+          #- runs-on: macos-latest
+          #  backend: vulkan
           #- runs-on: macos-latest
           #  backend: opengl
-          #  rust-version: msrv
     steps:
       - uses: actions/checkout@v4
+      - if: github.event_name != 'release' && github.event_name != 'workflow_dispatch'
+        uses: Swatinem/rust-cache@v2
       - uses: taiki-e/install-action@v2
         with: { tool: 'just' }
-      - uses: Swatinem/rust-cache@v2
-      - run: just env-info
-      - if: startsWith(matrix.runs-on, 'ubuntu')
-        name: Install linux dependencies
-        run: |
-          sudo apt-get update
-          # Make sure this list matches README
-          sudo apt-get install -y \
-            build-essential \
-            libcurl4-openssl-dev \
-            libglfw3-dev \
-            libjpeg-dev \
-            libpng-dev \
-            libsqlite3-dev \
-            libuv1-dev \
-            libwebp-dev \
-            libz-dev
-          if [ "${{ matrix.backend }}" = "opengl" ]; then
-            sudo apt-get install -y libopengl0
-          elif [ "${{ matrix.backend }}" = "vulkan" ]; then
-            sudo apt-get install -y mesa-vulkan-drivers glslang-dev
-          fi
-      - if: startsWith(matrix.runs-on, 'macos')
-        name: Install macOS dependencies
-        run: |
-          # Install dependencies via Homebrew
-          brew install \
-            curl \
-            glfw \
-            jpeg \
-            libpng \
-            sqlite \
-            libuv \
-            webp \
-            zlib
-          # For Vulkan on macOS, install MoltenVK
-          if [ "${{ matrix.backend }}" = "vulkan" ]; then
-            brew install molten-vk vulkan-headers
-          fi
-      # Setup rust version
-      - if: matrix.rust-version == 'stable'
-        run: rustup update stable && rustup default stable
-      - if: matrix.rust-version == 'msrv'
-        name: Read MSRV
+      - run: just env-info install-dependencies '${{ matrix.backend }}'
+      - name: Read MSRV
         id: msrv
-        run: echo "value=$(sed -nE 's/^rust-version[[:space:]]*=[[:space:]]*"([^"]+)".*/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
-      - if: matrix.rust-version == 'msrv'
-        name: Install Rust
+        run: echo "value=$(just get-msrv)" >> $GITHUB_OUTPUT
+      - name: Install MSRV Rust ${{ steps.msrv.outputs.value }}
         uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ steps.msrv.outputs.value }}
-      # Run tests
-      - run: just ci-test ${{ matrix.backend }}
+      - run: just ci_mode=0 ci-test-msrv ${{ matrix.backend }}  # Ignore warnings in MSRV
```

Comment:
`ci_mode=0 ci-test-msrv` seems rather weird.

Why not push this down to `ci-test-msrv`?

Reply:
because apparently a just recipe cannot modify a global just variable - so it has to be done as a param.  Note that `just var1=val1 var2=val2  task  param1 param2` is a common pattern for running just